tests/all/js_globals/Reflect.rs

```diff
@@ -0,0 +1,703 @@
+#![allow(non_snake_case)]
+
+use project;
+
+#[test]
+fn apply() {
+    project()
+        .file(
+            "src/lib.rs",
+            r#"
+            #![feature(proc_macro, wasm_custom_section)]
+
+            extern crate wasm_bindgen;
+            use wasm_bindgen::prelude::*;
+            use wasm_bindgen::js;
+
+            #[wasm_bindgen]
+            pub fn apply(target: &js::Function, this_argument: &JsValue, arguments_list: &js::Array) -> JsValue {
+                let result = js::Reflect::apply(target, this_argument, arguments_list);
+                let result = match result {
```

Comment:
Could you avoid name shadowing? This code could be a bit more concise with:
```rust
match result {
    Ok(val) => val,
    Err(_err) => "TypeError".into(),
}
```

`match` is an expression that returns value - note that I omitted the semicolon at the end :)

Reply:
This I actually took from [here](https://github.com/rustwasm/wasm-bindgen/blob/2694dd4dee28e36981e026e60d5bdf141f51562d/tests/all/js_globals/Number.rs#L75) without thinking to much. You are absolutely right. Maybe those tests could use some refactoring there as well for future reference.

Reply:
great 👍